src/components/RandomDog.js

```diff
@@ -1,20 +1,47 @@
-import React from 'react';
-import './RandomDog.css';
+import React from "react";
+import "./RandomDog.css";
 
 class RandomDog extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      randomDog: [],
+      savedPhotos: []
+    };
+  }
 
-	render() {
-		return (
-			<div className="RandomDog">
-				<h2 className="RandomDog-title">Random Dogs</h2>
-				<img className="RandomDog-image" src="http://via.placeholder.com/300x300" />
-				<p>
-					<button className="RandomDog-button">Save Image</button>
-					<button className="RandomDog-button">Next Dog</button>
-				</p>
-			</div>
-		)
-	}
+  componentDidMount() {
+    const randomDogLink = "https://dog.ceo/api/breeds/image/random";
+    fetch(randomDogLink)
+      .then(data => data.json())
+      .then(randompic => {
+        this.setState({ randomDog: randompic.message });
+      });
+  }
+
+  saveImage = () => {
+    this.setState({ savedPhotos: this.state.randomDog });
+  };
+
+  getNextImage = () => {
+    return this.componentDidMount();
```

Comment:
In React, the `componentDidMount` method is only expected to be fired once. Running it again might cause confusion.

Instead, try to perform the functions of `componentDidMount` in the `getNextImage` method, and call that from `componentDidMount`. It should look something like this:

```
componentDidMount() {
  this.getNextImage();
}

getNextImage = () => {
  // ... get your random dog image and set state
}
```

Reply:
I have changed it too.. Please check.. Thanks..